aws/data_source_aws_ssm_patch_baseline.go

```diff
@@ -0,0 +1,123 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func dataSourceAwsSsmPatchBaseline() *schema.Resource {
+	return &schema.Resource{
+		Read: dataAwsSsmPatchBaselineRead,
+		Schema: map[string]*schema.Schema{
+			"owner": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"name_prefix": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"default_baseline": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+			"operating_system": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			// Computed values
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataAwsSsmPatchBaselineRead(d *schema.ResourceData, meta interface{}) error {
+	ssmconn := meta.(*AWSClient).ssmconn
+
+	filters := []*ssm.PatchOrchestratorFilter{
+		{
+			Key: aws.String("OWNER"),
+			Values: []*string{
+				aws.String(d.Get("owner").(string)),
+			},
+		},
+	}
+
+	if v, ok := d.GetOk("name_prefix"); ok {
+		filters = append(filters, &ssm.PatchOrchestratorFilter{
+			Key: aws.String("NAME_PREFIX"),
+			Values: []*string{
+				aws.String(v.(string)),
+			},
+		})
+	}
+
+	params := &ssm.DescribePatchBaselinesInput{
+		Filters: filters,
+	}
+
+	log.Printf("[DEBUG] Reading DescribePatchBaselines: %s", params)
+
+	resp, err := ssmconn.DescribePatchBaselines(params)
+
+	if err != nil {
+		return fmt.Errorf("Error describing SSM PatchBaselines: %s", err)
+	}
+
+	var filteredBaselines []*ssm.PatchBaselineIdentity
+	if v, ok := d.GetOk("operating_system"); ok {
+		for _, baseline := range resp.BaselineIdentities {
+			if v.(string) == *baseline.OperatingSystem {
+				filteredBaselines = append(filteredBaselines, baseline)
+			}
+		}
+	}
+
+	if v, ok := d.GetOk("default_baseline"); ok {
+		var ln int
+		for _, baseline := range filteredBaselines {
+			if v.(bool) == *baseline.DefaultBaseline {
+				filteredBaselines[ln] = baseline
+				ln++
+			}
+		}
+		filteredBaselines = filteredBaselines[:ln]
```

Comment:
True we can just break on the first one - my intention was to validate the response we get back from AWS to check if multiple default baselines were returned - as opposed to assuming there would only be one. I'll use your suggestion since it is an unlikely edge case.